---
uint32 val1
uint32 val2